src/commands/mod.rs

```diff
@@ -1841,6 +1844,57 @@ fn edit_description(
     Ok(text_util::complete_newline(description))
 }
 
+fn edit_sparse(
+    cwd: &Path,
+    workspace_root: &Path,
+    repo_path: &Path,
+    sparse: &[RepoPath],
+    settings: &UserSettings,
+) -> Result<Vec<RepoPath>, CommandError> {
+    let file_path = (|| -> Result<_, io::Error> {
+        let mut file = tempfile::Builder::new()
+            .prefix("editor-")
+            .suffix(".jjsparse")
+            .tempfile_in(repo_path)?;
+        for sparse_path in sparse {
+            file.write_all(
+                file_util::relative_path(cwd, &sparse_path.to_fs_path(workspace_root))
```

Comment:
I wonder if we should make these patterns be relative to the workspace root. I think this is good for now since `jj sparse set --add/--remove` take cwd-relative paths. We should revisit that later. (Not asking for any changes in this PR.)

Reply:
Ah, I agree. Changed to be relative to workspace root. I wrote the awkward `file_util::relative_path(workspace_root, &sparse_path.to_fs_path(workspace_root))` instead of just `sparse_path.to_internal_file_string()` because I'm not sure they're equivalent.

Reply:
They are not quite equivalent because `to_internal_file_string()` will use `/` as separator regardless of platform. I think what you have is good for now since it makes the path format the same for paths provided on the command line (to `jj sparse set --add`) and as paths provided in the editor.